multiqc/modules/qualimap/QM_BamQC.py

```diff
@@ -116,16 +122,32 @@ def parse_genome_results(self, f):
         if line.startswith(">>>>>>>"):
             section = line[8:]
         elif section:
-            for k, r in regexes.get(section, {}).items():
-                r_search = re.search(r, line)
-                if r_search:
-                    if r"\d" in r:
-                        try:
-                            d[k] = float(r_search.group(1).replace(",", ""))
-                        except ValueError:
-                            d[k] = r_search.group(1)
-                    else:
-                        d[k] = r_search.group(1)
+            for key, (regex, regex_type) in regexes.get(section, {}).items():
+                regex = rf"{regex}\s*=\s*({regex_by_type[regex_type]})"
+                m = re.search(regex, line)
+                if not m:
+                    continue
+                val = m.group(1)
+                val = re.sub(r"\xa0", "", val)
+                if regex_type == int:
+                    d[key] = int(re.sub(r"[,\.]", "", val))
+                elif regex_type == float:
+                    # need to determine either dot or comma is either a decimal or thousands separator
+                    if val.count(".") == 1 and val.count(",") == 0:
+                        if len(val.split(".")[-1]) < 3:  # 1.12
```

Comment:
Does it make sense to figure this out once per file, rather than repeatedly - would that be more safe? Or is it not needed?

I've written similar code elsewhere, eg. [here](https://github.com/MultiQC/MultiQC/blob/d26af5e19c2ebde308bac339fb1b0c3a2dafa99a/multiqc/modules/picard/HsMetrics.py#L114-L121). (Basically all the Java tools have this problem).

Reply:
Optimized to determine the format once per file. Assumes that "rate" metrics like GC% or error rate will never have a thousand separator.

Reply:
Great 👍🏻